src/Microsoft.AspNetCore.SignalR/DefaultHubLifetimeManager.cs

```diff
@@ -106,6 +106,9 @@ public override Task InvokeUserAsync(string userId, string methodName, object[]
 
         public override Task OnConnectedAsync(HubConnectionContext connection)
         {
+            // Set the hub groups feature
+            connection.Features.Set<IHubGroupsFeature>(new HubGroupsFeature());
```

Comment:
Why not set it when creating the HubConnectionContext? This is creating a pit of failure (NRE or races) in cases where people want to add the connection to a group in their OnConnectedAsync which is common.